runtime/metricsview/executor_summary.go

```diff
@@ -0,0 +1,302 @@
+)
+
+// SummaryResult is the statistics for a single metrics view.
+type SummaryResult struct {
+	DefaultTimeDimension DimensionSummary   `json:"default_time_dimension,omitempty"`
+	Dimensions           []DimensionSummary `json:"dimensions"`
+}
+
+// DimensionSummary provides statistics for a single dimension in the metrics view.
+type DimensionSummary struct {
+	Name         string `json:"name"`
+	DataType     string `json:"data_type"`
+	ExampleValue any    `json:"example_value,omitempty"`
+	HasNulls     bool   `json:"has_nulls,omitempty"`
+	MinValue     any    `json:"min_value,omitempty"`
+	MaxValue     any    `json:"max_value,omitempty"`
+}
+
+// Summary provides statistics for all dimensions and measures in the metrics view.
+func (e *Executor) Summary(ctx context.Context) (*SummaryResult, error) {
+	if !e.security.CanAccess() {
+		return nil, runtime.ErrForbidden
+	}
+
+	dimensions := make([]*runtimev1.MetricsViewSpec_Dimension, 0, DefaultSummaryLimit)
+	timeDimensions := make([]*runtimev1.MetricsViewSpec_Dimension, 0, 2)
+
+	// Track the default time dimension name to ensure it's included
+	defaultTimeDimName := e.metricsView.TimeDimension
+
+	for _, dim := range e.metricsView.Dimensions {
+		// Skip dimensions that the user cannot access first
+		if !e.security.CanAccessField(dim.Name) {
+			continue
+		}
+
+		// If we have reached the limit, stop adding dimensions
+		if len(dimensions)+len(timeDimensions) >= DefaultSummaryLimit {
+			break
+		}
+
+		// Check if this dimension is a time dimension (has timestamp data type)
+		isTimeDimension := false
+		if dim.DataType != nil {
+			switch dim.DataType.Code {
+			case runtimev1.Type_CODE_TIMESTAMP, runtimev1.Type_CODE_DATE, runtimev1.Type_CODE_TIME:
+				isTimeDimension = true
+			}
+		}
+
+		if isTimeDimension {
+			timeDimensions = append(timeDimensions, dim)
+		} else {
+			dimensions = append(dimensions, dim)
+		}
+	}
+
+	// Ensure the default time dimension is included even if it's not in the dimensions list
+	defaultTimeDimFound := false
+	if defaultTimeDimName != "" {
+		for _, dim := range timeDimensions {
+			if dim.Name == defaultTimeDimName {
+				defaultTimeDimFound = true
+				break
+			}
+		}
+	}
+
+	// Get the default time range for the metrics view using the default time dimension
+	// This will be used for time range filtering in other queries
+	var defaultTimeRange TimestampsResult
+	var err error
+	if defaultTimeDimName != "" {
+		defaultTimeRange, err = e.Timestamps(ctx, defaultTimeDimName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get default time range for dimension %q: %w", defaultTimeDimName, err)
+		}
+	}
+
+	timeDimensionSummaries := make([]DimensionSummary, 0, len(timeDimensions)+1)
+
+	// Process each time dimension found in the dimensions list
+	for _, dim := range timeDimensions {
+		timeRange, err := e.Timestamps(ctx, dim.Name)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get time range for dimension %s: %w", dim.Name, err)
+		}
```

Comment:
If the `defaultTimeDimName` is present in `timeDimensions`, then it will end up calling `e.Timestamps` twice for it.